community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/planner/logical/CardinalityCostModel.scala

```diff
@@ -67,18 +89,24 @@ case class CardinalityCostModel(cardinality: CardinalityModel) extends CostModel
     case CartesianProduct(lhs, rhs) =>
       apply(lhs, input) + cardinality(lhs, input) * apply(rhs, input)
 
-    case Apply(lhs, rhs) =>
-      val newInputCardinality = cardinality(lhs, input)
-      val newInput = input.withCardinality(newInputCardinality)
-
+    case ApplyVariants(lhs, rhs) =>
+      val lhsCardinality = cardinality(lhs, input)
       val lCost = apply(lhs, input)
-      val rCost = apply(rhs, newInput)
-      lCost + rCost
+      val rCost = apply(rhs, input)
+
+      lCost + lhsCardinality * rCost
 
-    case OuterHashJoin(_, lhs, rhs) =>
+    case HashJoin(lhs, rhs) =>
       val lCost = apply(lhs, input)
       val rCost = apply(rhs, input)
-      lCost + rCost
+      val PROBE_BUILD_COST = CPU_BOUND
+      val PROBE_SEARCH_COST = CPU_BOUND * .5
```

Comment:
shouldn't these constant move to a static context since they are not dependent on input?